Mathlib/Probability/Kernel/IntegralCompProd.lean

```diff
@@ -99,7 +99,7 @@ theorem hasFiniteIntegral_compProd_iff ⦃f : β × γ → E⦄ (h1f : StronglyM
     rw [← and_congr_right_iff, and_iff_right_of_imp h1]
   rw [this]
   · intro h2f; rw [lintegral_congr_ae]
-    refine' h2f.mp _; apply eventually_of_forall; intro x hx; dsimp only
+    refine' h2f.mp _; filter_upwards with x hx
```

Comment:
```suggestion
    filter_upwards [h2f] with x hx
```
IMHO, this is the true power of `filter_upwards`; when you're using it without a [...] argument the gain over `eventually_of_forall` is minor. (The rest of this proof can also golfed considerably but that's not the point of this PR.)